### DISCUSSION TITLE
Access to plugin data and implementation of plugin methods.

### DISCUSSION THREAD

**Discussion by joseto1298** [mdziekon/octoprint-spoolman#60] [Ideas]
I'm developing a plugin to store print information and I need to access the plugin data.
- Filament used up during printing.
- Filament used up during printing.
- System to subtract used up filament in case of unexpected failure

I'm developing that, I want to know if you would implement it in your plugins so I can use it in my plugins.

**Comment by mdziekon** [comment]
I was in the middle of re-reviewing your Pull Request which I assume was supposed to unlock that functionality for your plugin. I don't know why you've removed your branch, but if you want me to continue, you can revert your removal and I'll go back into re-reviewing it once I have spare time.

However, what you've described here does not necessarily go hand in hand with the previously proposed backup system (it does supplement it for sure, but it sounds like partial usage event could be emitted without the backup system being enabled), so I would strongly suggest keeping these functionalities separate, if possible. I did suggest that in my first review, but as far as I could see, you did not comply with that comment, neither did challenge it (if for whatever reason it was necessary to do it that way).

**Reply by joseto1298** [reply]
I'm going to create a pull with changes and improvements with events, so that I can access it from other plugins and the backup system.
According to your specifications.

Thanks

**Comment by joseto1298** [comment]
I would like to synchronize the pull so I can finish and publish my plugin
